[
    {
        "number": 482,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-9wx4-h78v-vm56",
            "cve_id": "CVE-2024-35195",
            "summary": "Requests `Session` object does not verify requests after making first request with verify=False",
            "description": "When making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same origin will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool.\n\n### Remediation\nAny of these options can be used to remediate the current issue, we highly recommend upgrading as the preferred mitigation.\n\n* Upgrade to `requests>=2.32.0`.\n* For `requests<2.32.0`, avoid setting `verify=False` for the first request to a host while using a Requests Session.\n* For `requests<2.32.0`, call `close()` on `Session` objects to clear existing connections if `verify=False` is used.\n\n### Related Links\n* https://github.com/psf/requests/pull/6655",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-9wx4-h78v-vm56",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-35195",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56"
                },
                {
                    "url": "https://github.com/psf/requests/pull/6655"
                },
                {
                    "url": "https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-35195"
                },
                {
                    "url": "https://github.com/advisories/GHSA-9wx4-h78v-vm56"
                }
            ],
            "published_at": "2024-05-20T20:15:00Z",
            "updated_at": "2024-05-20T22:07:29Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "requests"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.32.0",
                    "first_patched_version": {
                        "identifier": "2.32.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N",
                "score": 5.6
            },
            "cwes": [
                {
                    "cwe_id": "CWE-670",
                    "name": "Always-Incorrect Control Flow Implementation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.32.0",
            "first_patched_version": {
                "identifier": "2.32.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/482",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/482",
        "created_at": "2024-05-24T01:35:27Z",
        "updated_at": "2024-05-24T01:35:27Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 481,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Werkzeug"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-2g68-c3qc-8985",
            "cve_id": "CVE-2024-34069",
            "summary": "Werkzeug debugger vulnerable to remote execution when interacting with attacker controlled domain",
            "description": "The debugger in affected versions of Werkzeug can allow an attacker to execute code on a developer's machine under some circumstances. This requires the attacker to get the developer to interact with a domain and subdomain they control, and enter the debugger PIN, but if they are successful it allows access to the debugger even if it is only running on localhost. This also requires the attacker to guess a URL in the developer's application that will trigger the debugger.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-2g68-c3qc-8985",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34069",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34069"
                },
                {
                    "url": "https://github.com/advisories/GHSA-2g68-c3qc-8985"
                }
            ],
            "published_at": "2024-05-06T14:21:27Z",
            "updated_at": "2024-05-06T16:17:35Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Werkzeug"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 3.0.3",
                    "first_patched_version": {
                        "identifier": "3.0.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-352",
                    "name": "Cross-Site Request Forgery (CSRF)"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Werkzeug"
            },
            "severity": "high",
            "vulnerable_version_range": "< 3.0.3",
            "first_patched_version": {
                "identifier": "3.0.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/481",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/481",
        "created_at": "2024-05-24T01:35:27Z",
        "updated_at": "2024-05-24T01:35:27Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 480,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h75v-3vvj-5mfj",
            "cve_id": "CVE-2024-34064",
            "summary": "Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for the previous GHSA-h5c8-rqwp-cp95 CVE-2024-22195 only addressed spaces but not other characters.\n\nAccepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h75v-3vvj-5mfj",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34064",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj"
                },
                {
                    "url": "https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34064"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h75v-3vvj-5mfj"
                }
            ],
            "published_at": "2024-05-06T14:20:59Z",
            "updated_at": "2024-05-06T16:17:32Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Jinja2"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.1.4",
                    "first_patched_version": {
                        "identifier": "3.1.4"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N",
                "score": 5.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-79",
                    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.1.4",
            "first_patched_version": {
                "identifier": "3.1.4"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/480",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/480",
        "created_at": "2024-05-24T01:35:27Z",
        "updated_at": "2024-05-24T01:35:27Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 479,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "flask-cors"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-84pr-m4jr-85g5",
            "cve_id": "CVE-2024-1681",
            "summary": "flask-cors vulnerable to log injection when the log level is set to debug",
            "description": "corydolphin/flask-cors is vulnerable to log injection when the log level is set to debug. An attacker can inject fake log entries into the log file by sending a specially crafted GET request containing a CRLF sequence in the request path. This vulnerability allows attackers to corrupt log files, potentially covering tracks of other attacks, confusing log post-processing tools, and forging log entries. The issue is due to improper output neutralization for logs.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-84pr-m4jr-85g5",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-1681",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-1681"
                },
                {
                    "url": "https://huntr.com/bounties/25a7a0ba-9fa2-4777-acb6-03e5539bb644"
                },
                {
                    "url": "https://github.com/corydolphin/flask-cors/blob/40acc8092332dfed4bb54d7a4f89a6d479466de7/flask_cors/extension.py#L194"
                },
                {
                    "url": "https://github.com/advisories/GHSA-84pr-m4jr-85g5"
                }
            ],
            "published_at": "2024-04-19T21:31:08Z",
            "updated_at": "2024-05-07T13:28:05Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "flask-cors"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "<= 4.0.0",
                    "first_patched_version": {
                        "identifier": "4.0.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:N",
                "score": 5.3
            },
            "cwes": [
                {
                    "cwe_id": "CWE-117",
                    "name": "Improper Output Neutralization for Logs"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "flask-cors"
            },
            "severity": "medium",
            "vulnerable_version_range": "<= 4.0.0",
            "first_patched_version": {
                "identifier": "4.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/479",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/479",
        "created_at": "2024-05-24T01:35:27Z",
        "updated_at": "2024-05-24T01:35:27Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 478,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "idna"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-jjg7-2v4v-x38h",
            "cve_id": "CVE-2024-3651",
            "summary": "Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode",
            "description": "### Impact\nA specially crafted argument to the `idna.encode()` function could consume significant resources. This may lead to a denial-of-service.\n\n### Patches\nThe function has been refined to reject such strings without the associated resource consumption in version 3.7.\n\n### Workarounds\nDomain names cannot exceed 253 characters in length, if this length limit is enforced prior to passing the domain to the `idna.encode()` function it should no longer consume significant resources. This is triggered by arbitrarily large inputs that would not occur in normal usage, but may be passed to the library assuming there is no preliminary input validation by the higher-level application.\n\n### References\n* https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-jjg7-2v4v-x38h",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-3651",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/kjd/idna/security/advisories/GHSA-jjg7-2v4v-x38h"
                },
                {
                    "url": "https://github.com/advisories/GHSA-jjg7-2v4v-x38h"
                }
            ],
            "published_at": "2024-04-11T21:32:40Z",
            "updated_at": "2024-04-11T21:32:40Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "idna"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.7",
                    "first_patched_version": {
                        "identifier": "3.7"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 6.2
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "idna"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.7",
            "first_patched_version": {
                "identifier": "3.7"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/478",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/478",
        "created_at": "2024-05-24T01:35:27Z",
        "updated_at": "2024-05-24T01:35:27Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 477,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-44wm-f244-xhp3",
            "cve_id": "CVE-2024-28219",
            "summary": "Pillow buffer overflow vulnerability",
            "description": "In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-44wm-f244-xhp3",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-28219",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-28219"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/2a93aba5cfcf6e241ab4f9392c13e3b74032c061"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M"
                },
                {
                    "url": "https://github.com/advisories/GHSA-44wm-f244-xhp3"
                }
            ],
            "published_at": "2024-04-03T03:30:30Z",
            "updated_at": "2024-04-20T05:06:18Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 10.3.0",
                    "first_patched_version": {
                        "identifier": "10.3.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
                "score": 6.7
            },
            "cwes": [
                {
                    "cwe_id": "CWE-120",
                    "name": "Buffer Copy without Checking Size of Input ('Classic Buffer Overflow')"
                },
                {
                    "cwe_id": "CWE-676",
                    "name": "Use of Potentially Dangerous Function"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 10.3.0",
            "first_patched_version": {
                "identifier": "10.3.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/477",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/477",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 476,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-3f63-hfp8-52jq",
            "cve_id": "CVE-2023-50447",
            "summary": "Arbitrary Code Execution in Pillow",
            "description": "Pillow through 10.1.0 allows PIL.ImageMath.eval Arbitrary Code Execution via the environment parameter, a different vulnerability than CVE-2022-22817 (which was about the expression parameter).",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-3f63-hfp8-52jq",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-50447",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-50447"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/45c726fd4daa63236a8f3653530f297dc87b160a"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#security"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2024/01/20/1"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/releases"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html"
                },
                {
                    "url": "https://devhub.checkmarx.com/cve-details/CVE-2023-50447"
                },
                {
                    "url": "https://duartecsantos.github.io/2023-01-02-CVE-2023-50447"
                },
                {
                    "url": "https://duartecsantos.github.io/2024-01-02-CVE-2023-50447"
                },
                {
                    "url": "https://github.com/advisories/GHSA-3f63-hfp8-52jq"
                }
            ],
            "published_at": "2024-01-19T21:30:35Z",
            "updated_at": "2024-03-27T23:53:02Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.2.0",
                    "first_patched_version": {
                        "identifier": "10.2.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
                "score": 8.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-94",
                    "name": "Improper Control of Generation of Code ('Code Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.2.0",
            "first_patched_version": {
                "identifier": "10.2.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/476",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/476",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 475,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "jinja2"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h5c8-rqwp-cp95",
            "cve_id": "CVE-2024-22195",
            "summary": "Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing spaces. XML/HTML attributes cannot contain spaces, as each would then be interpreted as a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. Note that accepting keys as user input is not common or a particularly intended use case of the `xmlattr` filter, and an application doing so should already be verifying what keys are provided regardless of this fix.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h5c8-rqwp-cp95",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-22195",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-22195"
                },
                {
                    "url": "https://github.com/pallets/jinja/commit/716795349a41d4983a9a4771f7d883c96ea17be7"
                },
                {
                    "url": "https://github.com/pallets/jinja/releases/tag/3.1.3"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h5c8-rqwp-cp95"
                }
            ],
            "published_at": "2024-01-11T15:20:48Z",
            "updated_at": "2024-01-27T03:30:20Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "jinja2"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.1.3",
                    "first_patched_version": {
                        "identifier": "3.1.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N",
                "score": 5.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-79",
                    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "jinja2"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.1.3",
            "first_patched_version": {
                "identifier": "3.1.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/475",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/475",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 474,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "werkzeug"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-hrfv-mqp8-q5rw",
            "cve_id": "CVE-2023-46136",
            "summary": "Werkzeug DoS: High resource usage when parsing multipart/form-data containing a large part with CR/LF character at the beginning",
            "description": "Werkzeug multipart data parser needs to find a boundary that may be between consecutive chunks. That's why parsing is based on looking for newline characters. Unfortunately, code looking for partial boundary in the buffer is written inefficiently, so if we upload a file that starts with CR or LF and then is followed by megabytes of data without these characters: all of these bytes are appended chunk by chunk into internal bytearray and lookup for boundary is performed on growing buffer.\n\nThis allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. The amount of RAM required can trigger an out of memory kill of the process. If many concurrent requests are sent continuously, this can exhaust or kill all available workers.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-hrfv-mqp8-q5rw",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-46136",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/b1916c0c083e0be1c9d887ee2f3d696922bfc5c1"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-46136"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/werkzeug/PYSEC-2023-221.yaml"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/f2300208d5e2a5076cbbb4c2aad71096fd040ef9"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20231124-0008/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-hrfv-mqp8-q5rw"
                }
            ],
            "published_at": "2023-10-25T14:22:59Z",
            "updated_at": "2023-12-03T05:04:51Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "werkzeug"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 3.0.0, < 3.0.1",
                    "first_patched_version": {
                        "identifier": "3.0.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "werkzeug"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.3.8",
                    "first_patched_version": {
                        "identifier": "2.3.8"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
                "score": 5.7
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                },
                {
                    "cwe_id": "CWE-787",
                    "name": "Out-of-bounds Write"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "werkzeug"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.3.8",
            "first_patched_version": {
                "identifier": "2.3.8"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/474",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/474",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 473,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-8ghj-p4vj-mr35",
            "cve_id": "CVE-2023-44271",
            "summary": "Pillow Denial of Service vulnerability",
            "description": "An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-8ghj-p4vj-mr35",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-44271",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-44271"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/pull/7244"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2023-227.yaml"
                },
                {
                    "url": "https://devhub.checkmarx.com/cve-details/CVE-2023-44271"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4"
                },
                {
                    "url": "https://github.com/advisories/GHSA-8ghj-p4vj-mr35"
                }
            ],
            "published_at": "2023-11-03T06:36:30Z",
            "updated_at": "2024-03-22T15:25:43Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 0, < 10.0.0",
                    "first_patched_version": {
                        "identifier": "10.0.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                },
                {
                    "cwe_id": "CWE-770",
                    "name": "Allocation of Resources Without Limits or Throttling"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 0, < 10.0.0",
            "first_patched_version": {
                "identifier": "10.0.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/473",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/473",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 472,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-g4mx-q9vg-27p4",
            "cve_id": "CVE-2023-45803",
            "summary": "urllib3's request body not stripped after redirect from 303 status changes request method to GET",
            "description": "urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 303 \"See Other\" after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although the behavior of removing the request body is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers.\n\nFrom [RFC 9110 Section 9.3.1](https://www.rfc-editor.org/rfc/rfc9110.html#name-get):\n\n> A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.\n\n## Affected usages\n\nBecause the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable.\n\nBoth of the following conditions must be true to be affected by this vulnerability:\n\n* If you're using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON)\n* The origin service is compromised and starts redirecting using 303 to a malicious peer or the redirected-to service becomes compromised.\n\n## Remediation\n\nYou can remediate this vulnerability with any of the following steps:\n\n* Upgrade to a patched version of urllib3 (v1.26.18 or v2.0.7)\n* Disable redirects for services that you aren't expecting to respond with redirects with `redirects=False`.\n* Disable automatic redirects with `redirects=False` and handle 303 redirects manually by stripping the HTTP request body.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-g4mx-q9vg-27p4",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-45803",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/4e50fbc5db74e32cabd5ccc1ab81fc103adfe0b3"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/b594c5ceaca38e1ac215f916538fb128e3526a36"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/releases/tag/1.26.18"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/releases/tag/2.0.7"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-45803"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9"
                },
                {
                    "url": "https://www.rfc-editor.org/rfc/rfc9110.html#name-get"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-212.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-g4mx-q9vg-27p4"
                }
            ],
            "published_at": "2023-10-17T20:15:25Z",
            "updated_at": "2023-11-06T16:24:46Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.0.0, < 2.0.7",
                    "first_patched_version": {
                        "identifier": "2.0.7"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 0, < 1.26.18",
                    "first_patched_version": {
                        "identifier": "1.26.18"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
                "score": 4.2
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 0, < 1.26.18",
            "first_patched_version": {
                "identifier": "1.26.18"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/472",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/472",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 471,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-56pw-mpj4-fxww",
            "cve_id": null,
            "summary": "Bundled libwebp in Pillow vulnerable",
            "description": "Pillow versions before v10.0.1 bundled libwebp binaries in wheels that are vulnerable to CVE-2023-5129 (previously CVE-2023-4863). Pillow v10.0.1 upgrades the bundled libwebp binary to v1.3.2.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-56pw-mpj4-fxww",
                    "type": "GHSA"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-4863"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-5129"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2023-175.yaml"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst#1001-2023-09-15"
                },
                {
                    "url": "https://github.com/advisories/GHSA-56pw-mpj4-fxww"
                }
            ],
            "published_at": "2023-10-05T00:06:58Z",
            "updated_at": "2023-10-05T00:06:58Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.0.1",
                    "first_patched_version": {
                        "identifier": "10.0.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": null,
                "score": 0.0
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.0.1",
            "first_patched_version": {
                "identifier": "10.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/471",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/471",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 470,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-j7hp-h8jx-5ppr",
            "cve_id": "CVE-2023-4863",
            "summary": "libwebp: OOB write in BuildHuffmanTable",
            "description": "Heap buffer overflow in libwebp allow a remote attacker to perform an out of bounds memory write via a crafted HTML page. ",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-j7hp-h8jx-5ppr",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-4863",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-4863"
                },
                {
                    "url": "https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html"
                },
                {
                    "url": "https://crbug.com/1479274"
                },
                {
                    "url": "https://github.com/qnighy/libwebp-sys2-rs/pull/21"
                },
                {
                    "url": "https://github.com/qnighy/libwebp-sys2-rs/commit/4560c473a76ec8bd8c650f19ddf9d7a44f719f8b"
                },
                {
                    "url": "https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a"
                },
                {
                    "url": "https://bugzilla.suse.com/show_bug.cgi?id=1215231"
                },
                {
                    "url": "https://en.bandisoft.com/honeyview/history/"
                },
                {
                    "url": "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863"
                },
                {
                    "url": "https://news.ycombinator.com/item?id=37478403"
                },
                {
                    "url": "https://rustsec.org/advisories/RUSTSEC-2023-0060.html"
                },
                {
                    "url": "https://security-tracker.debian.org/tracker/CVE-2023-4863"
                },
                {
                    "url": "https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/"
                },
                {
                    "url": "https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/"
                },
                {
                    "url": "https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39823"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39825"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39826"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39827"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39828"
                },
                {
                    "url": "https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/"
                },
                {
                    "url": "https://github.com/webmproject/libwebp/releases/tag/v1.3.2"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/"
                },
                {
                    "url": "https://rustsec.org/advisories/RUSTSEC-2023-0061.html"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202309-05"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5496"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5497"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5498"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/"
                },
                {
                    "url": "https://blog.isosceles.com/the-webp-0day/"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/21/4"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/3"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/4"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/5"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/6"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/7"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/8"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/26/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/26/7"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/2"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/4"
                },
                {
                    "url": "https://github.com/jaredforth/webp/pull/30"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/pull/7395"
                },
                {
                    "url": "https://github.com/jaredforth/webp/commit/9d4c56e63abecc777df71c702503c3eaabd7dcbc"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.0.1.html#security"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20230929-0011/"
                },
                {
                    "url": "https://github.com/ImageMagick/ImageMagick/discussions/6664"
                },
                {
                    "url": "https://github.com/dlemstra/Magick.NET/releases/tag/13.3.0"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202401-10"
                },
                {
                    "url": "https://sethmlarson.dev/security-developer-in-residence-weekly-report-16"
                },
                {
                    "url": "https://www.bentley.com/advisories/be-2023-0001/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-j7hp-h8jx-5ppr"
                }
            ],
            "published_at": "2023-09-12T15:30:20Z",
            "updated_at": "2024-01-08T15:38:39Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "libwebp-sys2"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.1.8",
                    "first_patched_version": {
                        "identifier": "0.1.8"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "libwebp-sys"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.9.3",
                    "first_patched_version": {
                        "identifier": "0.9.3"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 22.0.0, < 22.3.24",
                    "first_patched_version": {
                        "identifier": "22.3.24"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 24.0.0, < 24.8.3",
                    "first_patched_version": {
                        "identifier": "24.8.3"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 25.0.0, < 25.8.1",
                    "first_patched_version": {
                        "identifier": "25.8.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 26.0.0, < 26.2.1",
                    "first_patched_version": {
                        "identifier": "26.2.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 27.0.0-beta.1, < 27.0.0-beta.2",
                    "first_patched_version": {
                        "identifier": "27.0.0-beta.2"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "SkiaSharp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2.0.0, < 2.88.6",
                    "first_patched_version": {
                        "identifier": "2.88.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "go",
                        "name": "github.com/chai2010/webp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 1.0.0",
                    "first_patched_version": null
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.0.1",
                    "first_patched_version": {
                        "identifier": "10.0.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "webp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.2.6",
                    "first_patched_version": {
                        "identifier": "0.2.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-hdri-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-openmp-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 8.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-787",
                    "name": "Out-of-bounds Write"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.0.1",
            "first_patched_version": {
                "identifier": "10.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/470",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/470",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 469,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-v845-jxx5-vc9f",
            "cve_id": "CVE-2023-43804",
            "summary": "`Cookie` HTTP header isn't stripped on cross-origin redirects",
            "description": "urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly.\n\nUsers **must** handle redirects themselves instead of relying on urllib3's automatic redirects to achieve safe processing of the `Cookie` header, thus we decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Using an affected version of urllib3 (patched in v1.26.17 and v2.0.6)\n* Using the `Cookie` header on requests, which is mostly typical for impersonating a browser.\n* Not disabling HTTP redirects\n* Either not using HTTPS or for the origin server to redirect to a malicious origin.\n\n## Remediation\n\n* Upgrading to at least urllib3 v1.26.17 or v2.0.6\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Cookie` header.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-v845-jxx5-vc9f",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-43804",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-43804"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-192.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-v845-jxx5-vc9f"
                }
            ],
            "published_at": "2023-10-02T23:27:05Z",
            "updated_at": "2023-11-04T00:30:22Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.0.0, < 2.0.6",
                    "first_patched_version": {
                        "identifier": "2.0.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.26.17",
                    "first_patched_version": {
                        "identifier": "1.26.17"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N",
                "score": 5.9
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 1.26.17",
            "first_patched_version": {
                "identifier": "1.26.17"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/469",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/469",
        "created_at": "2024-05-24T01:35:26Z",
        "updated_at": "2024-05-24T01:35:26Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 468,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-98vv-pw6r-q6q4",
            "cve_id": "CVE-2021-23437",
            "summary": "Uncontrolled Resource Consumption in pillow",
            "description": "The package pillow 5.2.0 and before 8.3.2 are vulnerable to Regular Expression Denial of Service (ReDoS) via the getrgb function.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-98vv-pw6r-q6q4",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2021-23437",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-23437"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/9e08eb8f78fdfd2f476e1b20b7cf38683754866b"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/8.3.2.html"
                },
                {
                    "url": "https://snyk.io/vuln/SNYK-PYTHON-PILLOW-1319443"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202211-10"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/RNSG6VFXTAROGF7ACYLMAZNQV4EJ6I2C"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VKRCL7KKAKOXCVD7M6WC5OKFGL4L3SJT"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RNSG6VFXTAROGF7ACYLMAZNQV4EJ6I2C"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VKRCL7KKAKOXCVD7M6WC5OKFGL4L3SJT"
                },
                {
                    "url": "https://github.com/advisories/GHSA-98vv-pw6r-q6q4"
                }
            ],
            "published_at": "2021-09-07T23:08:10Z",
            "updated_at": "2024-04-22T21:11:33Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 5.2.0, < 8.3.2",
                    "first_patched_version": {
                        "identifier": "8.3.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-125",
                    "name": "Out-of-bounds Read"
                },
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 5.2.0, < 8.3.2",
            "first_patched_version": {
                "identifier": "8.3.2"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/468",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/468",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 467,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-xqr8-7jwr-rhp7",
            "cve_id": "CVE-2023-37920",
            "summary": "Removal of e-Tugra root certificate",
            "description": "Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\n e-Tugra's root certificates are being removed pursuant to an investigation prompted by reporting of security issues in their systems. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A).",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-xqr8-7jwr-rhp7",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-37920",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7"
                },
                {
                    "url": "https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-37920"
                },
                {
                    "url": "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/certifi/PYSEC-2023-135.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-xqr8-7jwr-rhp7"
                }
            ],
            "published_at": "2023-07-25T14:43:53Z",
            "updated_at": "2023-11-09T05:05:47Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "certifi"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
                    "first_patched_version": {
                        "identifier": "2023.7.22"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-345",
                    "name": "Insufficient Verification of Data Authenticity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
            "first_patched_version": {
                "identifier": "2023.7.22"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/467",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/467",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 466,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tornado"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-qppv-j76h-2rpx",
            "cve_id": null,
            "summary": "Tornado vulnerable to HTTP request smuggling via improper parsing of `Content-Length` fields and chunk lengths",
            "description": "## Summary\nTornado interprets `-`, `+`, and `_` in chunk length and `Content-Length` values, which are not allowed by the HTTP RFCs. This can result in request smuggling when Tornado is deployed behind certain proxies that interpret those non-standard characters differently. This is known to apply to older versions of haproxy, although the current release is not affected.\n\n## Details\nTornado uses the `int` constructor to parse the values of `Content-Length` headers and chunk lengths in the following locations:\n### `tornado/http1connection.py:445`\n```python3\n            self._expected_content_remaining = int(headers[\"Content-Length\"])\n```\n### `tornado/http1connection.py:621`\n```python3\n                content_length = int(headers[\"Content-Length\"])  # type: Optional[int]\n```\n### `tornado/http1connection.py:671`\n```python3\n            chunk_len = int(chunk_len_str.strip(), 16)\n```\nBecause `int(\"0_0\") == int(\"+0\") == int(\"-0\") == int(\"0\")`, using the `int` constructor to parse and validate strings that should contain only ASCII digits is not a good strategy. \n\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-qppv-j76h-2rpx",
                    "type": "GHSA"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tornadoweb/tornado/security/advisories/GHSA-qppv-j76h-2rpx"
                },
                {
                    "url": "https://github.com/tornadoweb/tornado/commit/b7a5dd29bb02950303ae96055082c12a1ea0a4fe"
                },
                {
                    "url": "https://github.com/advisories/GHSA-qppv-j76h-2rpx"
                }
            ],
            "published_at": "2023-08-14T21:34:17Z",
            "updated_at": "2023-09-07T20:25:28Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tornado"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 6.3.3",
                    "first_patched_version": {
                        "identifier": "6.3.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": null,
                "score": 0.0
            },
            "cwes": [
                {
                    "cwe_id": "CWE-444",
                    "name": "Inconsistent Interpretation of HTTP Requests ('HTTP Request/Response Smuggling')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tornado"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 6.3.3",
            "first_patched_version": {
                "identifier": "6.3.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/466",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/466",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 465,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Pygments"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-mrwq-x4v8-fh7p",
            "cve_id": "CVE-2022-40896",
            "summary": "Pygments vulnerable to ReDoS",
            "description": "A ReDoS issue was discovered in `pygments/lexers/smithy.py` in Pygments until 2.15.0 via SmithyLexer.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-mrwq-x4v8-fh7p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-40896",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-40896"
                },
                {
                    "url": "https://github.com/pygments/pygments/blob/master/pygments/lexers/smithy.py#L61"
                },
                {
                    "url": "https://github.com/pygments/pygments/commit/97eb3d5ec7c1b3ea4fcf9dee30a2309cf92bd194"
                },
                {
                    "url": "https://github.com/pygments/pygments/commit/dd52102c38ebe78cd57748e09f38929fd283ad04"
                },
                {
                    "url": "https://github.com/pygments/pygments/commit/fdf182a7af85b1deeeb637ca970d31935e7c9d52"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pygments/PYSEC-2023-117.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/EZGMXALE3HSP4OXC7UUWIKX3OXKZDTY3"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VUZO4BQCIY2S2KZYHERQMKURB7AHXDBO"
                },
                {
                    "url": "https://pypi.org/project/Pygments"
                },
                {
                    "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages-part-2"
                },
                {
                    "url": "https://github.com/advisories/GHSA-mrwq-x4v8-fh7p"
                }
            ],
            "published_at": "2023-07-19T15:30:26Z",
            "updated_at": "2024-03-28T13:28:18Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Pygments"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.15.0",
                    "first_patched_version": {
                        "identifier": "2.15.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
                "score": 5.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-434",
                    "name": "Unrestricted Upload of File with Dangerous Type"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Pygments"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.15.0",
            "first_patched_version": {
                "identifier": "2.15.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/465",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/465",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 464,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-9hxf-ppjv-w6rq",
            "cve_id": "CVE-2023-32732",
            "summary": "gRPC connection termination issue",
            "description": "gRPC contains a vulnerability whereby a client can cause a termination of connection between a HTTP2 proxy and a gRPC server: a base64 encoding error for `-bin` suffixed headers will result in a disconnection by the gRPC server, but is typically allowed by HTTP2 proxies. We recommend upgrading beyond the commit in https://github.com/grpc/grpc/pull/32309.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-9hxf-ppjv-w6rq",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-32732",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-32732"
                },
                {
                    "url": "https://github.com/grpc/grpc/pull/32309"
                },
                {
                    "url": "https://github.com/grpc/grpc/releases/tag/v1.53.1"
                },
                {
                    "url": "https://github.com/rubysec/ruby-advisory-db/blob/master/gems/grpc/CVE-2023-32732.yml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/37IDNVY5AWVH7JDMM2SDTL24ZPPZJNSY/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VWE44J5FG7THHL7XVEVTNIGEYBNKJBLL/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-9hxf-ppjv-w6rq"
                }
            ],
            "published_at": "2023-07-06T21:15:08Z",
            "updated_at": "2023-11-11T05:06:16Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "maven",
                        "name": "io.grpc:grpc-protobuf"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "grpcio"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rubygems",
                        "name": "grpc"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L",
                "score": 5.3
            },
            "cwes": [
                {
                    "cwe_id": "CWE-440",
                    "name": "Expected Behavior Violation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 1.53.0",
            "first_patched_version": {
                "identifier": "1.53.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/464",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/464",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 463,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-6628-q6j9-w8vg",
            "cve_id": "CVE-2023-1428",
            "summary": "gRPC Reachable Assertion issue",
            "description": "There exists an vulnerability causing an abort() to be called in gRPC. \nThe following headers cause gRPC's C++ implementation to abort() when called via http2:\n\nte: x (x != trailers)\n\n:scheme: x (x != http, https)\n\ngrpclb_client_stats: x (x == anything)\n\nOn top of sending one of those headers, a later header must be sent that gets the total header size past 8KB. We recommend upgrading past git commit 2485fa94bd8a723e5c977d55a3ce10b301b437f8 or v1.53 and above.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-6628-q6j9-w8vg",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-1428",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-1428"
                },
                {
                    "url": "https://github.com/grpc/grpc/commit/2485fa94bd8a723e5c977d55a3ce10b301b437f8"
                },
                {
                    "url": "https://github.com/grpc/grpc/issues/33463"
                },
                {
                    "url": "https://github.com/rubysec/ruby-advisory-db/blob/master/gems/grpc/CVE-2023-1428.yml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-6628-q6j9-w8vg"
                }
            ],
            "published_at": "2023-07-06T21:15:08Z",
            "updated_at": "2023-11-10T05:00:50Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "maven",
                        "name": "io.grpc:grpc-protobuf"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "grpcio"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rubygems",
                        "name": "grpc"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-617",
                    "name": "Reachable Assertion"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "severity": "high",
            "vulnerable_version_range": "< 1.53.0",
            "first_patched_version": {
                "identifier": "1.53.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/463",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/463",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 462,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "joblib"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-6hrg-qmvc-2xh8",
            "cve_id": "CVE-2022-21797",
            "summary": "joblib vulnerable to arbitrary code execution",
            "description": "The package joblib from 0 and before 1.2.0 is vulnerable to Arbitrary Code Execution via the `pre_dispatch` flag in `Parallel()` class due to the `eval()` statement.",
            "severity": "critical",
            "identifiers": [
                {
                    "value": "GHSA-6hrg-qmvc-2xh8",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-21797",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-21797"
                },
                {
                    "url": "https://github.com/joblib/joblib/issues/1128"
                },
                {
                    "url": "https://github.com/joblib/joblib/pull/1321"
                },
                {
                    "url": "https://github.com/joblib/joblib/commit/b90f10efeb670a2cc877fb88ebb3f2019189e059"
                },
                {
                    "url": "https://security.snyk.io/vuln/SNYK-PYTHON-JOBLIB-3027033"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BVOMMW37OXZWU2EV5ONAAS462IQEHZOF/"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2022/11/msg00020.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MJ5XTJS6OKJRRVXWFN5J67K3BYPEOBDF/"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/03/msg00027.html"
                },
                {
                    "url": "https://github.com/joblib/joblib/pull/1327"
                },
                {
                    "url": "https://github.com/joblib/joblib/pull/1352"
                },
                {
                    "url": "https://github.com/joblib/joblib/commit/6638b9e9711ad1ebbf6dd95aa7cee0dca9897b42"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/joblib/PYSEC-2022-288.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/BVOMMW37OXZWU2EV5ONAAS462IQEHZOF/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/MJ5XTJS6OKJRRVXWFN5J67K3BYPEOBDF/"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202401-01"
                },
                {
                    "url": "https://github.com/advisories/GHSA-6hrg-qmvc-2xh8"
                }
            ],
            "published_at": "2022-09-27T00:00:22Z",
            "updated_at": "2024-04-22T21:11:28Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "joblib"
                    },
                    "severity": "critical",
                    "vulnerable_version_range": "< 1.2.0",
                    "first_patched_version": {
                        "identifier": "1.2.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
                "score": 9.8
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "joblib"
            },
            "severity": "critical",
            "vulnerable_version_range": "< 1.2.0",
            "first_patched_version": {
                "identifier": "1.2.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/462",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/462",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 461,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-cfgp-2977-2fmm",
            "cve_id": "CVE-2023-32731",
            "summary": "Connection confusion in gRPC",
            "description": "When gRPC HTTP2 stack raised a header size exceeded error, it skipped parsing the rest of the HPACK frame. This caused any HPACK table mutations to also be skipped, resulting in a desynchronization of HPACK tables between sender and receiver. If leveraged, say, between a proxy and a backend, this could lead to requests from the proxy being interpreted as containing headers from different proxy clients - leading to an information leak that can be used for privilege escalation or data exfiltration. We recommend upgrading beyond the commit contained in  https://github.com/grpc/grpc/pull/32309",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-cfgp-2977-2fmm",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-32731",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-32731"
                },
                {
                    "url": "https://github.com/grpc/grpc/pull/32309"
                },
                {
                    "url": "https://github.com/grpc/grpc/pull/33005"
                },
                {
                    "url": "https://github.com/grpc/grpc/issues/33463"
                },
                {
                    "url": "https://github.com/rubysec/ruby-advisory-db/blob/master/gems/grpc/CVE-2023-32731.yml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-cfgp-2977-2fmm"
                }
            ],
            "published_at": "2023-07-05T19:12:51Z",
            "updated_at": "2023-11-10T05:00:35Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "maven",
                        "name": "io.grpc:grpc-protobuf"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "grpcio"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rubygems",
                        "name": "grpc"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 1.53.0",
                    "first_patched_version": {
                        "identifier": "1.53.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:N/A:H",
                "score": 7.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-440",
                    "name": "Expected Behavior Violation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "grpcio"
            },
            "severity": "high",
            "vulnerable_version_range": "< 1.53.0",
            "first_patched_version": {
                "identifier": "1.53.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/461",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/461",
        "created_at": "2024-05-24T01:35:25Z",
        "updated_at": "2024-05-24T01:35:25Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 460,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tornado"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-hj3f-6gcp-jg8j",
            "cve_id": "CVE-2023-28370",
            "summary": "Open redirect in Tornado",
            "description": "Open redirect vulnerability in Tornado versions 6.3.1 and earlier allows a remote unauthenticated attacker to redirect a user to an arbitrary web site and conduct a phishing attack by having user access a specially crafted URL.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-hj3f-6gcp-jg8j",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-28370",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-28370"
                },
                {
                    "url": "https://github.com/tornadoweb/tornado/releases/tag/v6.3.2"
                },
                {
                    "url": "https://jvn.jp/en/jp/JVN45127776/"
                },
                {
                    "url": "https://github.com/tornadoweb/tornado/commit/32ad07c54e607839273b4e1819c347f5c8976b2f"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/tornado/PYSEC-2023-75.yaml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-hj3f-6gcp-jg8j"
                }
            ],
            "published_at": "2023-05-25T12:30:16Z",
            "updated_at": "2023-11-12T05:01:11Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tornado"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 6.3.2",
                    "first_patched_version": {
                        "identifier": "6.3.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N",
                "score": 6.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-601",
                    "name": "URL Redirection to Untrusted Site ('Open Redirect')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tornado"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 6.3.2",
            "first_patched_version": {
                "identifier": "6.3.2"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/460",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/460",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 459,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-j8r2-6x86-q33q",
            "cve_id": "CVE-2023-32681",
            "summary": "Unintended leak of Proxy-Authorization header in requests",
            "description": "### Impact\n\nSince Requests v2.3.0, Requests has been vulnerable to potentially leaking `Proxy-Authorization` headers to destination servers, specifically during redirects to an HTTPS origin. This is a product of how `rebuild_proxies` is used to recompute and [reattach the `Proxy-Authorization` header](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/sessions.py#L319-L328) to requests when redirected. Note this behavior has _only_ been observed to affect proxied requests when credentials are supplied in the URL user information component (e.g. `https://username:password@proxy:8080`).\n\n**Current vulnerable behavior(s):**\n\n1. HTTP → HTTPS: **leak**\n2. HTTPS → HTTP: **no leak**\n3. HTTPS → HTTPS: **leak**\n4. HTTP → HTTP: **no leak**\n\nFor HTTP connections sent through the proxy, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the `Proxy-Authorization` header must be sent in the CONNECT request as the proxy has no visibility into further tunneled requests. This results in Requests forwarding the header to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate those credentials.\n\nThe reason this currently works for HTTPS connections in Requests is the `Proxy-Authorization` header is also handled by urllib3 with our usage of the ProxyManager in adapters.py with [`proxy_manager_for`](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/adapters.py#L199-L235). This will compute the required proxy headers in `proxy_headers` and pass them to the Proxy Manager, avoiding attaching them directly to the Request object. This will be our preferred option going forward for default usage.\n\n### Patches\nStarting in Requests v2.31.0, Requests will no longer attach this header to redirects with an HTTPS destination. This should have no negative impacts on the default behavior of the library as the proxy credentials are already properly being handled by urllib3's ProxyManager.\n\nFor users with custom adapters, this _may_ be potentially breaking if you were already working around this behavior. The previous functionality of `rebuild_proxies` doesn't make sense in any case, so we would encourage any users impacted to migrate any handling of Proxy-Authorization directly into their custom adapter.\n\n### Workarounds\nFor users who are not able to update Requests immediately, there is one potential workaround.\n\nYou may disable redirects by setting `allow_redirects` to `False` on all calls through Requests top-level APIs. Note that if you're currently relying on redirect behaviors, you will need to capture the 3xx response codes and ensure a new request is made to the redirect destination.\n```\nimport requests\nr = requests.get('http://github.com/', allow_redirects=False)\n```\n\n### Credits\n\nThis vulnerability was discovered and disclosed by the following individuals.\n\nDennis Brinkrolf, Haxolot (https://haxolot.com/)\nTobias Funke, (tobiasfunke93@gmail.com)",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-j8r2-6x86-q33q",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-32681",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q"
                },
                {
                    "url": "https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-32681"
                },
                {
                    "url": "https://github.com/psf/requests/releases/tag/v2.31.0"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/requests/PYSEC-2023-74.yaml"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202309-08"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ"
                },
                {
                    "url": "https://github.com/advisories/GHSA-j8r2-6x86-q33q"
                }
            ],
            "published_at": "2023-05-22T20:36:32Z",
            "updated_at": "2024-03-27T20:09:55Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "requests"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.3.0, < 2.31.0",
                    "first_patched_version": {
                        "identifier": "2.31.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N",
                "score": 6.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 2.3.0, < 2.31.0",
            "first_patched_version": {
                "identifier": "2.31.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/459",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/459",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 458,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "flask"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-m2qf-hxjv-5gpq",
            "cve_id": "CVE-2023-30861",
            "summary": "Flask vulnerable to possible disclosure of permanent session cookie due to missing Vary: Cookie header",
            "description": "When all of the following conditions are met, a response containing data intended for one client may be cached and subsequently sent by a proxy to other clients. If the proxy also caches `Set-Cookie` headers, it may send one client's `session` cookie to other clients. The severity depends on the application's use of the session, and the proxy's behavior regarding cookies. The risk depends on _all_ these conditions being met.\n\n1. The application must be hosted behind a caching proxy that does not strip cookies or ignore responses with cookies.\n2. The application sets [`session.permanent = True`](https://flask.palletsprojects.com/en/2.3.x/api/#flask.session.permanent).\n2. The application does not access or modify the session at any point during a request.\n4. [`SESSION_REFRESH_EACH_REQUEST`](https://flask.palletsprojects.com/en/2.3.x/config/#SESSION_REFRESH_EACH_REQUEST) is enabled (the default).\n5. The application does not set a `Cache-Control` header to indicate that a page is private or should not be cached.\n\nThis happens because vulnerable versions of Flask only set the `Vary: Cookie` header when the session is accessed or modified, not when it is refreshed (re-sent to update the expiration) without being accessed or modified.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-m2qf-hxjv-5gpq",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-30861",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/flask/security/advisories/GHSA-m2qf-hxjv-5gpq"
                },
                {
                    "url": "https://github.com/pallets/flask/commit/70f906c51ce49c485f1d355703e9cc3386b1cc2b"
                },
                {
                    "url": "https://github.com/pallets/flask/releases/tag/2.3.2"
                },
                {
                    "url": "https://github.com/pallets/flask/commit/afd63b16170b7c047f5758eb910c416511e9c965"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-30861"
                },
                {
                    "url": "https://github.com/pallets/flask/releases/tag/2.2.5"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/flask/PYSEC-2023-62.yaml"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5442"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20230818-0006/"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/08/msg00024.html"
                },
                {
                    "url": "https://github.com/advisories/GHSA-m2qf-hxjv-5gpq"
                }
            ],
            "published_at": "2023-05-01T19:22:20Z",
            "updated_at": "2023-11-06T05:03:38Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "flask"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2.3.0, < 2.3.2",
                    "first_patched_version": {
                        "identifier": "2.3.2"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "flask"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.2.5",
                    "first_patched_version": {
                        "identifier": "2.2.5"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-539",
                    "name": "Use of Persistent Cookies Containing Sensitive Information"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "flask"
            },
            "severity": "high",
            "vulnerable_version_range": "< 2.2.5",
            "first_patched_version": {
                "identifier": "2.2.5"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/458",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/458",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 457,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h6q3-vv32-2cq5",
            "cve_id": "CVE-2022-41894",
            "summary": "Buffer overflow in `CONV_3D_TRANSPOSE` on TFLite",
            "description": "### Impact\nThe reference kernel of the [`CONV_3D_TRANSPOSE`](https://github.com/tensorflow/tensorflow/blob/091e63f0ea33def7ecad661a5ac01dcafbafa90b/tensorflow/lite/kernels/internal/reference/conv3d_transpose.h#L121) TensorFlow Lite operator wrongly increments the data_ptr when adding the bias to the result.\n\nInstead of `data_ptr += num_channels;` it should be `data_ptr += output_num_channels;` as if the number of input channels is different than the number of output channels, the wrong result will be returned and a buffer overflow will occur if num_channels > output_num_channels.\n\nAn attacker can craft a model with a specific number of input channels in a way similar to the attached example script. It is then possible to write specific values through the bias of the layer outside the bounds of the buffer. This attack only works if the reference kernel resolver is used in the interpreter (i.e. `experimental_op_resolver_type=tf.lite.experimental.OpResolverType.BUILTIN_REF` is used).\n```python\nimport tensorflow as tf\nmodel = tf.keras.Sequential(\n    [\n        tf.keras.layers.InputLayer(input_shape=(2, 2, 2, 1024), batch_size=1),\n        tf.keras.layers.Conv3DTranspose(\n            filters=8,\n            kernel_size=(2, 2, 2),\n            padding=\"same\",\n            data_format=\"channels_last\",\n        ),\n    ]\n)\n\nconverter = tf.lite.TFLiteConverter.from_keras_model(model)\ntflite_model = converter.convert()\n\ninterpreter = tf.lite.Interpreter(\n    model_content=tflite_model,\n    experimental_op_resolver_type=tf.lite.experimental.OpResolverType.BUILTIN_REF,\n)\n\ninterpreter.allocate_tensors()\ninterpreter.set_tensor(\n    interpreter.get_input_details()[0][\"index\"], tf.zeros(shape=[1, 2, 2, 2, 1024])\n)\ninterpreter.invoke()\n```\n\n### Patches\nWe have patched the issue in GitHub commit [72c0bdcb25305b0b36842d746cc61d72658d2941](https://github.com/tensorflow/tensorflow/commit/72c0bdcb25305b0b36842d746cc61d72658d2941).\n\nThe fix will be included in TensorFlow 2.11. We will also cherrypick this commit on TensorFlow 2.10.1, 2.9.3, and TensorFlow 2.8.4, as these are also affected and still in supported range.\n\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n\n### Attribution\nThis vulnerability has been reported by Thibaut Goetghebuer-Planchon, Arm Ltd.\n",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-h6q3-vv32-2cq5",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-41894",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-h6q3-vv32-2cq5"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-41894"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/commit/72c0bdcb25305b0b36842d746cc61d72658d2941"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/blob/091e63f0ea33def7ecad661a5ac01dcafbafa90b/tensorflow/lite/kernels/internal/reference/conv3d_transpose.h#L121"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h6q3-vv32-2cq5"
                }
            ],
            "published_at": "2022-11-21T20:44:24Z",
            "updated_at": "2023-04-10T19:02:13Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.8.4",
                    "first_patched_version": {
                        "identifier": "2.8.4"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2.9.0, < 2.9.3",
                    "first_patched_version": {
                        "identifier": "2.9.3"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2.10.0, < 2.10.1",
                    "first_patched_version": {
                        "identifier": "2.10.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
                "score": 7.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-120",
                    "name": "Buffer Copy without Checking Size of Input ('Classic Buffer Overflow')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 2.8.4",
            "first_patched_version": {
                "identifier": "2.8.4"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/457",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/457",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 456,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-fxgc-95xx-grvq",
            "cve_id": "CVE-2023-25661",
            "summary": "TensorFlow Denial of Service vulnerability",
            "description": "### Impact\nA malicious invalid input crashes a tensorflow model (Check Failed) and can be used to trigger a denial of service attack.\nTo minimize the bug, we built a simple single-layer TensorFlow model containing a Convolution3DTranspose layer, which works well with expected inputs and can be deployed in real-world systems. However, if we call the model with a malicious input which has a zero dimension, it gives Check Failed failure and crashes. \n```python\nimport tensorflow as tf\n\nclass MyModel(tf.keras.Model):\n    def __init__(self):\n        super().__init__()\n        self.conv = tf.keras.layers.Convolution3DTranspose(2, [3,3,3], padding=\"same\")\n        \n    def call(self, input):\n        return self.conv(input)\nmodel = MyModel() # Defines a valid model.\n\nx = tf.random.uniform([1, 32, 32, 32, 3], minval=0, maxval=0, dtype=tf.float32) # This is a valid input.\noutput = model.predict(x)\nprint(output.shape) # (1, 32, 32, 32, 2)\n\nx = tf.random.uniform([1, 32, 32, 0, 3], dtype=tf.float32) # This is an invalid input.\noutput = model(x) # crash\n```\nThis Convolution3DTranspose layer is a very common API in modern neural networks. The ML models containing such vulnerable components could be deployed in ML applications or as cloud services. This failure could be potentially used to trigger a denial of service attack on ML cloud services.\n\n### Patches\nWe have patched the issue in\n- GitHub commit [948fe6369a5711d4b4568ea9bbf6015c6dfb77e2](https://github.com/tensorflow/tensorflow/commit/948fe6369a5711d4b4568ea9bbf6015c6dfb77e2)\n- GitHub commit [85db5d07db54b853484bfd358c3894d948c36baf](https://github.com/keras-team/keras/commit/85db5d07db54b853484bfd358c3894d948c36baf).\n\nThe fix will be included in TensorFlow 2.12.0. We will also cherrypick this commit on TensorFlow 2.11.1\n\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-fxgc-95xx-grvq",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-25661",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-fxgc-95xx-grvq"
                },
                {
                    "url": "https://github.com/keras-team/keras/commit/85db5d07db54b853484bfd358c3894d948c36baf"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/commit/948fe6369a5711d4b4568ea9bbf6015c6dfb77e2"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-25661"
                },
                {
                    "url": "https://github.com/advisories/GHSA-fxgc-95xx-grvq"
                }
            ],
            "published_at": "2023-03-27T21:05:10Z",
            "updated_at": "2023-09-01T15:23:50Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-cpu"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
                "score": 6.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-20",
                    "name": "Improper Input Validation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.11.1",
            "first_patched_version": {
                "identifier": "2.11.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/456",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/456",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 455,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-93vr-9q9m-pj8p",
            "cve_id": "CVE-2023-25659",
            "summary": "TensorFlow vulnerable to Out-of-Bounds Read in DynamicStitch",
            "description": "### Impact\nIf the parameter `indices` for `DynamicStitch` does not match the shape of the parameter `data`, it can trigger an stack OOB read.\n\n```python\nimport tensorflow as tf\nfunc = tf.raw_ops.DynamicStitch\npara={'indices': [[0xdeadbeef], [405], [519], [758], [1015]], 'data': [[110.27793884277344], [120.29475402832031], [157.2418212890625], [157.2626953125], [188.45382690429688]]}\ny = func(**para)\n```\n\n### Patches\nWe have patched the issue in GitHub commit [ee004b18b976eeb5a758020af8880236cd707d05](https://github.com/tensorflow/tensorflow/commit/ee004b18b976eeb5a758020af8880236cd707d05).\n\nThe fix will be included in TensorFlow 2.12. We will also cherrypick this commit on TensorFlow 2.11.1.\n\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n\n### Attribution\nThis has been reported via Google OSS VRP.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-93vr-9q9m-pj8p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-25659",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-93vr-9q9m-pj8p"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/commit/ee004b18b976eeb5a758020af8880236cd707d05"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-25659"
                },
                {
                    "url": "https://github.com/advisories/GHSA-93vr-9q9m-pj8p"
                }
            ],
            "published_at": "2023-03-24T21:58:53Z",
            "updated_at": "2023-03-27T21:26:46Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-cpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-gpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-125",
                    "name": "Out-of-bounds Read"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 2.11.1",
            "first_patched_version": {
                "identifier": "2.11.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/455",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/455",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 454,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-qjqc-vqcf-5qvj",
            "cve_id": "CVE-2023-25660",
            "summary": "TensorFlow vulnerable to seg fault in `tf.raw_ops.Print`",
            "description": "### Impact\nWhen the parameter `summarize` of `tf.raw_ops.Print` is zero, the new method `SummarizeArray<bool>` will reference to a nullptr, leading to a seg fault.\n\n```python\nimport tensorflow as tf\n\ntf.raw_ops.Print(input =  tf.constant([1, 1, 1, 1],dtype=tf.int32),\n                            data =  [[False, False, False, False], [False], [False, False, False]],\n                            message =  'tmp/I',\n                            first_n = 100,\n                            summarize = 0)\n```\n\n### Patches\nWe have patched the issue in GitHub commit [6d423b8bcc9aa9f5554dc988c1c16d038b508df1](https://github.com/tensorflow/tensorflow/commit/6d423b8bcc9aa9f5554dc988c1c16d038b508df1).\n\nThe fix will be included in TensorFlow 2.12. We will also cherrypick this commit on TensorFlow 2.11.1.\n\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n\n### Attribution\nThis vulnerability has been reported by Yu Tian of Qihoo 360 AIVul Team\n",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-qjqc-vqcf-5qvj",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-25660",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-qjqc-vqcf-5qvj"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/commit/6d423b8bcc9aa9f5554dc988c1c16d038b508df1"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-25660"
                },
                {
                    "url": "https://github.com/advisories/GHSA-qjqc-vqcf-5qvj"
                }
            ],
            "published_at": "2023-03-24T21:58:44Z",
            "updated_at": "2023-03-27T21:27:24Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-cpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-gpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-476",
                    "name": "NULL Pointer Dereference"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 2.11.1",
            "first_patched_version": {
                "identifier": "2.11.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/454",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/454",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 453,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "manifest_path": "requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-7jvm-xxmr-v5cw",
            "cve_id": "CVE-2023-25662",
            "summary": "TensorFlow vulnerable to integer overflow in EditDistance",
            "description": "### Impact\nTFversion 2.11.0 //tensorflow/core/ops/array_ops.cc:1067 const Tensor* hypothesis_shape_t = c->input_tensor(2); std::vector<DimensionHandle> dims(hypothesis_shape_t->NumElements() - 1); for (int i = 0; i < dims.size(); ++i) { dims[i] = c->MakeDim(std::max(h_values(i), t_values(i))); }\n\nif hypothesis_shape_t is empty, hypothesis_shape_t->NumElements() - 1 will be integer overflow, and the it will deadlock\n```python\nimport tensorflow as tf\npara={\n    'hypothesis_indices': [[]],\n    'hypothesis_values': ['tmp/'],\n    'hypothesis_shape': [],\n    'truth_indices': [[]],\n    'truth_values': [''],\n    'truth_shape': [],\n    'normalize': False\n    }\ntf.raw_ops.EditDistance(**para)\n```\n\n### Patches\nWe have patched the issue in GitHub commit [08b8e18643d6dcde00890733b270ff8d9960c56c](https://github.com/tensorflow/tensorflow/commit/08b8e18643d6dcde00890733b270ff8d9960c56c).\n\nThe fix will be included in TensorFlow 2.12.0. We will also cherrypick this commit on TensorFlow 2.11.1\n\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n\n### Attribution\nThis vulnerability has been reported by r3pwnx",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-7jvm-xxmr-v5cw",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-25662",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-7jvm-xxmr-v5cw"
                },
                {
                    "url": "https://github.com/tensorflow/tensorflow/commit/08b8e18643d6dcde00890733b270ff8d9960c56c"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-25662"
                },
                {
                    "url": "https://github.com/advisories/GHSA-7jvm-xxmr-v5cw"
                }
            ],
            "published_at": "2023-03-24T21:58:31Z",
            "updated_at": "2023-03-27T21:58:28Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-cpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tensorflow-gpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 2.11.1",
                    "first_patched_version": {
                        "identifier": "2.11.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-190",
                    "name": "Integer Overflow or Wraparound"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tensorflow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 2.11.1",
            "first_patched_version": {
                "identifier": "2.11.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/Flight_Fare_Prediction/dependabot/alerts/453",
        "html_url": "https://github.com/fcas/Flight_Fare_Prediction/security/dependabot/453",
        "created_at": "2024-05-24T01:35:24Z",
        "updated_at": "2024-05-24T01:35:24Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    }
]